at_client/connections/atmonitorconnection.py

```diff
@@ -0,0 +1,222 @@
+                        key = str(event_data.get("key"))
+                        if "epochMillis" in event_data:
+                            self.last_received_time = int(event_data.get("epochMillis"))
+                        else:
+                            self.last_received_time = TimeUtil.current_time_millis()
+                        if uuid == "-1":
+                            event_type = AtEventType.STATS_NOTIFICATION
+                        elif operation == "update":
+                            if key.startswith(str(self.atsign.to_string) + ":shared_key@"):
+                                event_type = AtEventType.SHARED_KEY_NOTIFICATION
+                            else:
+                                event_type = AtEventType.UPDATE_NOTIFICATION
+                        elif operation == "delete":
+                            event_type = AtEventType.DELETE_NOTIFICATION
+                        else:
+                            event_type = AtEventType.MONITOR_EXCEPTION
+                            event_data["key"] = "__monitorException__"
+                            event_data["value"] = response.decode()
+                            event_data["exception"] = "Unknown notification operation " + str(operation)
+                    else:
+                        event_type = AtEventType.MONITOR_EXCEPTION
+                        event_data["key"] = "__monitorException__"
+                        event_data["value"] = response.decode()
+                        event_data["exception"] = "Malformed response from server"
+                except Exception as e:
+                    print(e)
+                    event_type = AtEventType.MONITOR_EXCEPTION
+                    event_data["key"] = "__monitorException__"
+                    event_data["value"] = response.decode()
+                    event_data["exception"] = str(e)
+                    traceback.print_exc()
+
+                at_event = AtEvent(event_type, event_data)
+                self.queue.put(at_event)
+                
+                should_be_running_lock.acquire(blocking=1)
+                entered = False
+            if not entered:
+                should_be_running_lock.release()
+                entered = False
+        except Exception as e:
+            traceback.print_exc()
+            should_be_running_lock.acquire(blocking=1)
+            if not self.should_be_running:
+                should_be_running_lock.release()
+            else:
+                should_be_running_lock.release()
+                print("Monitor failed to " + what + " : " + str(e))
+                traceback.print_exc()
+                print("Monitor ending. Monitor heartbeat thread should restart the monitor shortly")
+                self.disconnect()
+        finally:
+            running_lock.acquire(blocking=1)
+            self.running = False
+            running_lock.release()
+            
+            self.disconnect()
+            
+    def connect(self):
```

Comment:
```suggestion
    def _connect(self):
```

Should this be _connect?

Reply:
Yes, you are right. That should be changed